### ISSUE TITLE
Support for Lambda Layers

### ISSUE THREAD

**Issue by Vallo** [danieluhm2004/serverless-webpack-prisma#9]
Would be great to have support for Lambda Layers. This way we will ship both @prisma and .prisma together in a layer that will be used in the other Lambdas. With the current approach, each Lambda function is about 45MB in size. By using a Lambda Layer I have a single Layer 45MB in size, and each Lambda is about 3kb in size. 
The only benefit of this speeds up the deployment process, it does not speed up cold starts as the final size of the Lambda will be of all layers combined. But when having 50 Lambdas to deploy the difference in deploy speed is huge.

**Comment by gkawin** [comment]
@Vallo  Here is my workaround.

I use the Serverless Hook lifecycle [Serverless-scriptable-plugin](https://github.com/weixu365/serverless-scriptable-plugin) to solve this problem.

Here is my example on my `serverless.yaml`
```yaml
service: awesome-program
configValidationMode: error
frameworkVersion: "3"

resources: ${file(./config/resources.yaml):resources}
layers: ${file(./config/layers.yaml):layers}
functions: ${file(./config/functions.yaml):functions}

custom:
    webpack:
        webpackConfig: "./webpack.config.js"
        includeModules: true
    serverless-offline:
        # useChildProcesses: true
        ignoreJWTSignature: true
    s3: ${file(./config/configuration.yaml):s3-local.${self:provider.stage}}
    scriptable:
        hooks:
            package:initialize: sh ./.prisma-generate.sh
            after:webpack:package:packExternalModules: find .webpack/*/node_modules/.prisma .webpack/*/node_modules/@prisma -type f -name '*' -exec rm -rf {} \;
```

Here is my layers section
```yaml
layers:
    prisma:
        path: .prisma-layer
        description: "Layer for Prisma Clients"
        package:
            patterns:
                - "!nodejs/node_modules/@prisma/engines/*"
                - "!nodejs/node_modules/.prisma/**/libquery_engine-*"
                - "nodejs/node_modules/.prisma/**/libquery_engine-rhel-openssl-1.0.x.so.node"
                - "!nodejs/prisma/**"
```
The Scriptable section allows you to hook lifecycle while a Serverless is still deploying. So, I decided to use the Linux command to find and delete all of Prisma and related pieces of stuff in the bundled node_module, that appeared on the `.webpack` folder. After that, the Webpack does packing into a Zip file and delivered to the S3 bucket.

In conclusion, You should delete `@prisma` and `.prisma` folders in the `.webpack/<your_lambda_name>/node_modules` folder. Otherwise, your lambda will use in-function node_modules rather than AWS Lambda Layer.

I recommend you to read [this blog](https://dev.to/sabinthedev/managing-lambda-size-with-multiple-prisma-clients-4m8l?hmsr=joyk.com&utm_source=joyk.com&utm_medium=referral) first. This blog wrote by a Prisma contributor. He clarifies and guides you, that how to split the Prisma engine from AWS Lambda to AWS  Lambda Layer.
